runtime/doc/lsp.txt

```diff
@@ -60,48 +60,43 @@ options are not restored when the LSP client is stopped or detached.
   - To opt out of this use |gw| instead of gq, or set 'formatexpr' on LspAttach.
 - |K| is mapped to |vim.lsp.buf.hover()| unless |'keywordprg'| is customized or
   a custom keymap for `K` exists.
+- *crr* is mapped in Normal mode to |vim.lsp.buf.rename()| if no existing
+  keymap exists
+- *cra* is mapped in Normal mode to |vim.lsp.buf.code_action()| if no existing
+  keymap exists
+- |gr| is mapped in Normal mode to |vim.lsp.buf.references()| if no existing
+  keymap exists
+- CTRL-S is mapped in Insert mode to |vim.lsp.buf.signature_help()| if no
+  existing keymap exists *i_CTRL-S*
 
                                                         *lsp-defaults-disable*
 To override the above defaults, set or unset the options on |LspAttach|: >lua
+
     vim.api.nvim_create_autocmd('LspAttach', {
       callback = function(ev)
         vim.bo[ev.buf].formatexpr = nil
         vim.bo[ev.buf].omnifunc = nil
-        vim.keymap.del("n", "K", { buffer = ev.buf })
-      end,
-    })
-
-To use other LSP features like hover, rename, etc. you can set other keymaps
-on |LspAttach|. Example: >lua
-    vim.api.nvim_create_autocmd('LspAttach', {
-      callback = function(args)
-        vim.keymap.set('n', 'K', vim.lsp.buf.hover, { buffer = args.buf })
+        vim.keymap.del('n', 'K', { buffer = ev.buf })
+        vim.keymap.del('n', 'crr', { buffer = ev.buf })
+        vim.keymap.del('n', 'cra', { buffer = ev.buf })
+        vim.keymap.del('n', 'gr', { buffer = ev.buf })
+        vim.keymap.del('i', '<C-S>', { buffer = ev.buf })
       end,
     })
 
-The most common functions are:
-
-- |vim.lsp.buf.hover()|
-- |vim.lsp.buf.format()|
-- |vim.lsp.buf.references()|
-- |vim.lsp.buf.implementation()|
-- |vim.lsp.buf.code_action()|
-
+To use other LSP features, set keymaps on |LspAttach|. Not all language
+servers provide the same capabilities. To ensure you only set keymaps if the
+language server supports a feature, guard keymaps behind capability checks.
+Example: >lua
 
-Not all language servers provide the same capabilities. To ensure you only set
-keymaps if the language server supports a feature, you can guard the keymap
-calls behind capability checks:
->lua
     vim.api.nvim_create_autocmd('LspAttach', {
       callback = function(args)
-        local client = vim.lsp.get_client_by_id(args.data.client_id)
-        if client.server_capabilities.hoverProvider then
-          vim.keymap.set('n', 'K', vim.lsp.buf.hover, { buffer = args.buf })
+        if client.supports_method(ms.textDocument_implementation) then
```

Comment:
I love the use of the `Methods` module here; however I think there's still some discussion going on (see https://github.com/neovim/neovim/pull/24977) about its usage in examples.

Reply:
I think some of the arguments made in that PR don't apply to the example here, but it looks like the `ms` definition is missing and that would definitely trip up people.

Reply:
Updated in ac59371ef9aa2c979043bee574e576bdf62fb664